src/stage_node.cpp

```diff
@@ -279,10 +279,12 @@ int StageNode::callback_update_stage_world(Stg::World * world, StageNode * node)
     }
   }
 
+  // update obstacles
```

Comment:
@vjuenemann I slowly have the feeling that it might make sense to implemented the observer pattern to update the obstacles (see https://en.wikipedia.org/wiki/Observer_pattern). What do you think? If you have questions just let me know.

Reply:
An update function in the StageNode class would also be fine